src/acceptance-test/java/net/consensys/orion/acceptance/dsl/OrionFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.acceptance.dsl;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.orion.config.Config;
+import net.consensys.orion.enclave.sodium.FileKeyStore;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class OrionFactory {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final String libSodiumPath = "/usr/local/lib/libsodium.dylib";
+
+
+  public OrionNode create(final String nodeName, int keyCount) throws IOException {
+    return create(nodeName, keyCount, Collections.emptyList());
+  }
+
+  public OrionNode create(final String nodeName, final int keyCount, final List<OrionNode> bootnodes)
+      throws IOException {
+    final Path nodePath;
+    try {
+      nodePath = Files.createTempDirectory("orion");
+    } catch (final IOException e) {
+      LOG.error("Failed to create tmpdir for orion instance.");
+      throw e;
+    }
+
+    final Config config = Config.defaultConfig();
+    try {
+      final FileKeyStore fileKeyStore = new FileKeyStore(config);
+
+      final List<KeyDefinition> nodeKeys = Lists.newArrayList();
+      for (int i = 0; i < keyCount; i++) {
```

Comment:
Feels like this creation of 'keyCount' of generated keys should be done outside this factory and passed in as a set of keys to use?

Reply:
The keys are such an intrinsic part of the config, I didn't want to expose them outside of this creation (i.e. abstract all facets of config) (i.e. it all must appear under the same subdirectory etc.etc.)

I have moved key creation to a separate function for cleanliness - but making the user of this class create the keys for injection seemed to expose too much info ...